acceptance/new-user/join-button.test.js

```diff
@@ -0,0 +1,76 @@
+const puppeteer = require("puppeteer");
+const config = require("config");
+
+const URLS = require("../../constants/urls");
+
+const {JOIN } = URLS;
+
+
+let browser, page;
+
+
+/* time delay */
+const delay = (time) => {
+  return new Promise((res) => {
+    setTimeout(res, time);
+  });
+};
+
+/* confirm alerts */
+async function confirmAlerts(){
+        page.on('dialog', async dialog => {
+               await dialog.accept();
+               
+               delay(2000)
+               await page.waitForNavigation()
+
+               delay(4000)
+               await browser.close()
+        })
+}
+
+async function checkResponse(){
+ page.on("response", (response) => {
+            
+        if(response.url().endsWith("/users/self") && page.url().endsWith('/join')){
+        
+              switch (response.status()) {
+	           	case 401:
+			            delay(2000);
+			            confirmAlerts();
+			           break;
+		          case 200:
+			            browser.close();
+			         break;
+	          }
```

Comment:
can we have a default case as well . what if status code is other than 401 , 200 ? say 500

Reply:
what should I mention in default case? like if it is 500 then the browser should close?

Reply:
yeah we should close the browser otherwise the instance will remain open.